packages/flutter/test/widgets/scrollable_test.dart

```diff
@@ -368,6 +368,82 @@ void main() {
     expect(getScrollOffset(tester), 20.0);
   });
 
+  group('setCanDrag to false with active drag gesture: ', () {
+    Future<void> pumpTestWidget(WidgetTester tester, bool canDrag) async {
+      return await tester.pumpWidget(
+        MaterialApp(
+          home: CustomScrollView(
+            physics: canDrag ? const AlwaysScrollableScrollPhysics() : const NeverScrollableScrollPhysics(),
+            slivers: <Widget>[SliverToBoxAdapter(
+                child: SizedBox(
+                  height: 2000,
+                  child: GestureDetector(onTap: () {},),
+                ),
+            )],
+          ),
+        ),
+      );
+    }
+
+    testWidgets('hold', (WidgetTester tester) async {
+      // Regression test for https://github.com/flutter/flutter/issues/66816.
+      await pumpTestWidget(tester, true);
```

Comment:
nit: the test would be slightly more readable if "canDrag" was a required named parameter so you know what's set to true here.